spec/index.html

```diff
@@ -1493,3 +1503,221 @@ <h1>SharedArray ( ..._values_ )</h1>
+      <h1>Atomics.Mutex.tryLock ( _mutex_ [ , _unlockToken_ ] )</h1>
+      <p>It performs the following steps when called:</p>
+      <emu-alg>
+        1. If _mutex_ is not an Object, throw a *TypeError* exception.
+        1. If _mutex_ does not have a [[MutexWaiterList]] internal slot, throw a *TypeError* exception.
+        1. If _unlockToken_ not *undefined*, then
+          1. If _unlockToken_ is not an Object, throw a *TypeError* exception.
+          1. If _unlockToken_ does not have a [[LockedMutex]] internal slot, throw a *TypeError* exception.
+          1. If _unlockToken_.[[LockedMutex]] is not ~empty~, throw a *TypeError* exception.
+        1. Let _WL_ be _mutex_.[[MutexWaiterList]].
+        1. Perform EnterCriticalSection(_WL_).
+        1. If _mutex_.[[IsLocked]] is *false*, then
+          1. Set _mutex_.[[IsLocked]] to *true*.
+          1. Let _result_ be UnlockTokenCreateIfNeeded(_unlockToken_, _mutex_).
+        1. Else,
+          1. Let _result_ be *undefined*.
+        1. Perform LeaveCriticalSection(_WL_).
+        1. Return _result_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-atomics.mutex.withlock">
+      <h1>Atomics.Mutex.withLock ( _mutex_, _callback_ [ , _thisArg_ ] )</h1>
+      <p>It performs the following steps when called:</p>
+      <emu-alg>
+        1. If _mutex_ is not an Object, throw a *TypeError* exception.
+        1. If _mutex_ does not have a [[MutexWaiterList]] internal slot, throw a *TypeError* exception.
+        1. If IsCallable(_callback_) is *false*, throw a *TypeError* exception.
+        1. If AgentCanSuspend() is *false*, throw a *TypeError* exception.
+        1. Let _lockResult_ be LockMutex(_mutex_, +∞).
+        1. Assert: _lockResult_ is ~acquired~.
+        1. Let _callbackResult_ be ? Call(_callback_, _thisArg_, « »).
+        1. Perform UnlockMutex(_mutex_).
+        1. Return _callbackResult_.
+      </emu-alg>
+    </emu-clause>
+  </emu-clause>
+
+  <emu-clause id="sec-mutexunlocktokenprototype">
+    <h1>The %MutexUnlockTokenPrototype% Object</h1>
+    <p>The <dfn>%MutexUnlockTokenPrototype%</dfn> object:</p>
+    <ul>
+      <li>has a [[Prototype]] internal slot whose value is %Object.prototype%.</li>
+      <li>is an ordinary object.</li>
+      <li>has the following properties:</li>
+    </ul>
+
+    <emu-clause id="sec-%mutexunlocktokenprototype%.unlock">
+      <h1>%MutexUnlockTokenPrototype%.unlock ( )</h1>
+      <p>This function performs the following steps when called:</p>
+      <emu-alg>
+        1. Let _token_ be the *this* value.
+        1. If _token_ is not an Object, throw a *TypeError* exception.
+        1. If _token_ does not have a [[LockedMutex]] internal slot, throw a *TypeError* exception.
+        1. Let _mutex_ be _token_.[[LockedMutex]].
+        1. If _mutex_ is not ~empty~, then
+          1. Set _token_.[[LockedMutex]] to ~empty~.
+          1. Perform UnlockMutex(_mutex_).
+        1. Return *undefined*.
```

Comment:
Maybe return a boolean for whether the lock was locked? Could also add a getter for whether the lock is held. Not necessary at all but is a cheap thing to add.